main.go

```diff
@@ -202,3 +215,37 @@ func wrapAuthHandler(handler http.HandlerFunc, htpasswdFile string, host string)
 	authenticator := auth.NewBasicAuthenticator(host, secrets)
 	return auth.JustCheck(authenticator, handler)
 }
+
+func writePidFile(c *config.Config, accessLogger cache.Logger) error {
+	// create a "pid" directory under cache directory and clean up inactive pid file
+	pidPath := filepath.Join(c.Dir, "pid")
+	err := os.MkdirAll(pidPath, os.FileMode(0744))
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	err = filepath.Walk(pidPath, func(name string, fileInfo os.FileInfo, err error) error {
```

Comment:
Per the latest design doc, the pid file name is fixed to <bazel-remote version>.d, so no need to search for all pid file unless we want to kill the daemon of a different version.

Reply:
That is needed as the main purpose is to manage version upgrade. File name without .d looks simpler

Reply:
Per the design doc, for the same version of bazel-remote daemon, there is only one pid file name  <version>.d.  Do you want to kill the daemon of another version? Please note we are talking about the process of ANOTHER version, not the same version.